### PR TITLE
ZCS-11277: Upgrade Log4j version to 2.17.1

### PR DESCRIPTION
**Pull request by zimsuchitgupta**

### CODE REVIEW

File: ivy.xml [hunk 1]
```diff
@@ -11,7 +11,8 @@
   <dependency org="javax.servlet" name="javax.servlet-api" rev="3.1.0" />
   <dependency org="javax.mail" name="mail" rev="1.4.5" />
   <dependency org="org.dom4j" name="dom4j" rev="${dom4j.version}" />
-  <dependency org="log4j" name="log4j" rev="1.2.16" />
+  <dependency org="org.apache.logging.log4j" name="log4j-core" rev="${log4j.core.version}" />
```
Comment (silentsakky) [inline ivy.xml:14]: I guess client side zimlet code should not depend on log4j libs, we can remove this

**Review by shrutig0510 — APPROVED** [review]

**Review by gautamdg — APPROVED** [review]

**Review by abhishekdeshmukh2209 — APPROVED** [review]

File: ivy.xml [hunk 2]
```diff
@@ -11,7 +11,6 @@
   <dependency org="javax.servlet" name="javax.servlet-api" rev="3.1.0" />
   <dependency org="javax.mail" name="mail" rev="1.4.5" />
   <dependency org="org.dom4j" name="dom4j" rev="${dom4j.version}" />
-  <dependency org="log4j" name="log4j" rev="1.2.16" />
```
Comment (silentsakky) [inline ivy.xml:14]: better to revert this change, as this repo has some jsp and java files so maybe this dependency is needed here

Comment (suchitgupta11) [reply ivy.xml:14]: Added back log4j dependency

**Review by silentsakky — APPROVED** [review]
